core/src/main/java/org/mule/routing/correlation/EventCorrelator.java

```diff
@@ -125,21 +120,13 @@ public EventCorrelator(EventCorrelatorCallback callback,
         this.messageInfoMapping = messageInfoMapping;
         this.muleContext = muleContext;
         this.timeoutMessageProcessor = timeoutMessageProcessor;
-        this.persistentStores = persistentStores;
-        this.storePrefix = storePrefix;
         name = String.format("%s%s.event.correlator", ThreadNameHelper.getPrefix(muleContext),
                              flowConstruct.getName());
-        ObjectStoreManager objectStoreManager = muleContext.getRegistry().get(
-                MuleProperties.OBJECT_STORE_MANAGER);
-        expiredAndDispatchedGroups = (ListableObjectStore<Long>) objectStoreManager.getObjectStore(
-                storePrefix + ".expiredAndDispatchedGroups", persistentStores);
-        processedGroups = (ListableObjectStore<Long>) objectStoreManager.getObjectStore(storePrefix
-                                                                                        + ".processedGroups",
-                                                                                        persistentStores, MAX_PROCESSED_GROUPS, -1, 1000);
-        eventGroups = (ListableObjectStore<EventGroup>) objectStoreManager.getObjectStore(storePrefix
-                                                                                          + ".eventGroups",
-                                                                                          persistentStores);
         this.flowConstruct = flowConstruct;
+
+        this.expiredAndDispatchedGroups = expiredAndDispatchedGroups;
```

Comment:
In the previous version, EventCorrelator always created internal objectStores. In the new version, it always receives them injected. Why?